tests/scripts/analyze_outcomes.py

```diff
@@ -56,27 +56,27 @@ def hits(self):
         """
         return len(self.successes) + len(self.failures)
 
-def execute_reference_driver_tests(log: TestLog, ref_component, driver_component, \
-                                   outcome_file) -> TestLog:
+def execute_reference_driver_tests(results: Results, ref_component, driver_component, \
+                                   outcome_file) -> Results:
```

Comment:
Does the function really need to return a `Results` instance, considering it was passed by the caller? (Same question for all other functions that receive an instance of `Results` and return it.)